Sources/NostrSDK/DirectMessageEncrypting.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  DirectMessageEncrypting.swift
+//  
+//
+//  Created by Joel Klabo on 8/10/23.
+//
+
+import Foundation
+import secp256k1
+import CommonCrypto
+import CryptoKit
+
+public enum DirectMessageEncryptingError: Error {
+    case pubkeyParsing
+    case sharedSecretCreation
+    case encryptionError
+    case decryptionError
+    case missingValue
+}
+
+public protocol DirectMessageEncrypting {}
+public extension DirectMessageEncrypting {
+
+    func encrypt(content: String, privateKey: PrivateKey, publicKey: PublicKey) throws -> String {
```

Comment:
Can you add DocC comments, at least for these first two functions, that explain what they do and what the inputs and outputs are and how it could fail?